engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -1879,26 +1915,28 @@ def translateInvalid(invalid: AST.Invalid): Expression = {
     * Currently this only supports documentation comments, and not standarc
     * types of comments as they can't currently be represented.
     *
-    * @param tree the comment to transform
+    * @param doc the comment to transform
     * @return the [[IR]] representation of `comment`
     */
-  IR.Comment translateComment(Tree tree) {
-    return switch (tree) {
-      case Tree.ArgumentBlockApplication comment -> {
-        var doc = new StringBuilder();
-        var sep = "";
-        for (var l : comment.getArguments()) {
-          if (l.getExpression() != null) {
-            doc.append(sep);
-            doc.append(l.getExpression().codeRepr());
-            sep = "\n";
+  IR.Comment translateComment(Tree.Documented doc) {
+      var msg = new StringBuilder();
+      for (var t : doc.getDocumentation().getElements()) {
+        switch (t) {
+          case TextElement.Section s -> {
+            var txt = s.getText().codeRepr();
+            if (txt.startsWith("#")) {
+              txt = txt.substring(1);
+            }
+            if (msg.length() > 0) {
+              msg.append(" ");
+            }
```

Comment:
Whitespace handling in doc comments is currently the cause of IR-equivalence test failures in most source files. The problem is that in the new parser, I implement the [left-trim algorithm defined in the spec](https://github.com/enso-org/design/blob/wip/wd/enso-spec/epics/enso-spec-1.0/04.%20Expressions.md#inline-and-block-text-literals). The old parser didn't trim anything--the indentation that makes the block was included in the string. So, for compatibility we need to add the whitespace back in here, until the old parser is fully replaced and we can switch to the new semantics. Right now `Token` only provides whitespace info a source offsets; I'll add an accessor that makes this easier.

Reply:
I pushed a small commit integrating the new `getWhitespace`. For some reason trimming one space off every line seems to match the old parser's semantics.

Reply:
Thank you. Getting the indentation right with the old parser is tricky. Every help counts!